### PR TITLE
Initial support for go-task

### PR DESCRIPTION
**Pull request by GZGavinZhao**

### CODE REVIEW

**Comment by ReillyBrogan** [comment]
FWIW the Serpent one that we were working on is [here](https://github.com/snekpit/main/blob/main/Taskfile.dist.yml) if you want to take a look at it

File: Taskfile.yml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    dir: '{{ .USER_WORKING_DIR }}'
+    vars:
+      PROFILE: '{{ default "unstable-x86_64" .PROFILE }}'
+    deps:
+      - package-file
+    cmds:
+      - sudo solbuild build {{ .SPECFILE }} -p {{ .PROFILE }} {{ .CLI_ARGS }}
+
+  build-stable:
+    desc: Build the current package against the stable repo (do NOT use for official submissions!)
```
Comment (GZGavinZhao) [inline Taskfile.yml:35]: Do we want to use a [warning prompt](https://taskfile.dev/usage/#warning-prompts) instead?

File: Taskfile.yml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    aliases: [stable]
+    cmds:
+      - task: build
+        vars: 
+          PROFILE: 'main-x86_64'
+      - |
+        echo "=========================================================================="
```
[the remark this answers is not in the document]
Comment (ReillyBrogan) [reply Taskfile.yml:42]: Sure

**Comment by ReillyBrogan** [comment]
Something worth checking is if this works in fish shell as well. I believe that go-task runs commands in a subshell of the current shell so some bash constructs and syntax might not be valid unless you force the shell to be bash.

**Comment by GZGavinZhao** [comment]
@ReillyBrogan I use fish shell, so it definitely works on fish.

**Comment by EbonJaeger** [comment]
Trying to `publish` an update for real seems to have a problem:
image

The build then fails and no log can be found.

**Comment by EbonJaeger** [comment]
After the last commit, the following functions seem to work fine:
- `bump`
- `build`
- `publish`
  - `push` as a side-effect
- `republish`
- `clean`
- `pull`

**Pull request merged**